csrc/multidevice/utils.h

```diff
@@ -127,4 +129,44 @@ int64_t getShardedAxis(TensorView*);
 
 // Reorders a TensorView so that the DID parallelized axis are in front.
 void reorderDIDToFront(TensorView*);
+
+// Given a TensorView and the shape of a sharded tensor of which certain
+// dimensions are partially alloated, returns the global shape that'll be used
+// to bind to the TensorView's logical domain. This is to solve #3282 so we can
+// bind a sharded tensor to a TensorView that has a DID-parallel loop domain.
+//
+// For example, when `tv` is
+//   logical: iM, iN
+//   allocation: iDIDx{D}, iN/D, iM
+// and `sizes` is [2, 3], the returned shape will be [2, 3D]. This is because,
+// according to the allocation domain, iM is fully allocated and iN is sharded
+// and thus partially allocated.
+//
+// As a degenerate case, it's fine to call this function with a non-sharded
+// TensorView and tensor.
+//
+// Limitations:
+// - The function assumes that there are no Merges from logical to the
+// DID-parallel IterDomains in allocation. Otherwise, it's unclear which logical
+// dimension this DID-parallelization should be attributed to.
+// - The function assumes that all Splits from logical to the DID-parallel
+// IterDomains in allocation are even. This is because there are currently no
```

Comment:
nit: I don't think there ever will be a "way to pass the global shape". I would personally say that assuming that the splits are even is the canonical assumption (in the sense that among all assumptions possible, this is objectively the most natural one)

Reply:
Acknowledged -- I agree that assuming even splits is canonical. But I don't want to dismiss the possibility otherwise and therefore wrote "currently". I heard anecdotally people want uneven splits and nvFuser has been good at that for intra-GPU.